includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -2313,17 +2313,17 @@ private function deserialize_intention_object_from_array( array $intention_array
 	/**
 	 * Returns a formatted intention description.
 	 *
-	 * @param  int $order_id The order ID.
-	 * @return string        A formatted intention description.
+	 * @param  string $order_number The order number (might be different from the ID).
+	 * @return string               A formatted intention description.
 	 */
-	private function get_intent_description( int $order_id ): string {
+	private function get_intent_description( $order_number ): string {
```

Comment:
I think that we should include order ID if the order number is not set to maintain backward compatibility. For the orders that are created before this PR, the order number will not exist, so my suggestion is to refactor a code a bit on line 223 and 259 to include something like this: `$metadata['order_number'] ?? $metadata['order_id'] ?? 0`

Reply:
@zmaglica do you think that we need a fallback here? AFAICT the method will only be called for new orders, which need a description for their intent. FWIW I've checked all code paths I could think of, and the only place where we would not call `$order->get_order_number()` in advance is [here](https://github.com/Automattic/woocommerce-payments/pull/4275/files#diff-b541c0a287f40f82a1d0e43c52c7c7a2b67ae233f261655cbb437726fec83feeR276), and for that situation, there is a fallback to zero, in case the order does not exist.

Reply:
Makes sense. I will resolve this comment.